test/functional/autocmd/markset_spec.lua

```diff
@@ -0,0 +1,49 @@
+local t = require('test.testutil')
+local n = require('test.functional.testnvim')()
+
+local api = n.api
+local clear = n.clear
+local command = n.command
+local feed = n.feed
+local poke_eventloop = n.poke_eventloop
+local eval = n.eval
+
+local eq = t.eq
+
+describe('MarkSet', function()
```

Comment:
so far the tests are working? that is cool.

try adding a test that does insane things in the `MarkSet`, such as

- `:bwipeout!` on its buffer
- switches windows
- switches tabpage